bin/scripts/continious-data-population.js

```diff
@@ -0,0 +1,1183 @@
+        }
+    }
+];
+var app = null, template = null, pluginList = null;
+var users = [], user = null;
+var stats = {u: 0, s: 0, x: 0, d: 0, e: 0, r: 0, b: 0, c: 0, p: 0};
+
+var plugins = require('../../plugins/pluginManager.js'),
+    request = require('request'),
+    Chance = require('chance');
+
+
+function writeMsg(msg) {
+    process.stdout.write(JSON.stringify(msg) + '\n');
+    process.exit(1);
+}
+/**
+ * Checks the correctness of the required parameters.
+ **/
+async function readDbParameters() {
+    try {
+        const db = await plugins.dbConnection("countly");
+        app = await db.collection('apps').findOne({'_id': db.ObjectID(APP_ID)});
+        pluginList = await db.collection('plugins').find({}).toArray();
+        if (POPULATOR_TEMPLATE_ID.length > 18) {
+            template = await db.collection('populator_templates').findOne({'_id': db.ObjectID(POPULATOR_TEMPLATE_ID)});
+        }
+        else { // default template
+            defaultTemplates.forEach(function(item) {
+                if (item._id === POPULATOR_TEMPLATE_ID) {
+                    template = item;
+                }
+            });
+            if (!template) {
+                db.close();
+                writeMsg('INVALID POPULATOR_TEMPLATE_ID ', POPULATOR_TEMPLATE_ID);
+            }
+        }
+        if (!app) {
+            db.close();
+            writeMsg('INVALID APP_ID ', APP_ID);
+        }
+        if (!template) {
+            db.close();
+            writeMsg('INVALID POPULATOR_TEMPLATE_ID ', POPULATOR_TEMPLATE_ID);
+        }
+        db.close();
+    }
+    catch (error) {
+        clearInterval(interval);
+        writeMsg('There was an error while checking db parameters', error);
+    }
+}
+
+readDbParameters();
+
+var interval = setInterval(async function() {
+    users = [];
+    var userCount = getRandomInt(1, maxUserCount);
+    if (users.length === 0) {
+        for (var i = 0; i < userCount; i++) {
+            user = new getUser(template && template.up);
+            users.push(user);
+        }
+    }
+    else {
+        if (users.length !== maxUserCount) {
+            if (userCount === 1) {
+                user = new getUser(template && template.up);
+                users.push(user);
+            }
+        }
+    }
+
+    var upperBoundary = Math.floor(userCount / 2);
+
+    var indexes = [];
+    for (var i0 = 0; i0 < upperBoundary; i0++) {
+        indexes.push(getRandomInt(0, users.length - 1));
+    }
+
+    indexes.forEach(function(index) {
+        if (users[index].hasSession) {
+            if (users[index].eventsInSession > 9) {
+                users[index].eventsInSession = 0;
+                users[index].startSession(template);
+            }
+            else {
+                if (getRandomInt(1, 3) === 1) {
+                    users[index].addEvent(template, getRandomInt(1, template.events.length - 1));
+                    users[index].eventsInSession++;
+                }
```

Comment:
We have chosen  this user to have something. No need to filter out:
So I think we can remove that  if (getRandomInt(1, 3) === 1) {